### ISSUE TITLE
Add a WCS setter to NDData

### ISSUE THREAD

**Issue by mcara** [astropy/astropy#8192]
Once in a while one might need to replace a `NDData`'s `wcs` object with an "improved" version of the WCS. Current implementation does not allow doing this (unless I use `_wcs`). Would it be possible to relax this and add a setter for `wcs`?

**Comment by pllim** [comment]
cc @mwcraig and @MSeifert04

**Comment by mcara** [comment]
Here is the issue I am running into:
```python
In [32]: from astropy.wcs import WCS                                                                         

In [33]: from astropy.nddata import NDData                                                                   

In [34]: import numpy as np                                                                                  

In [35]: w1 = WCS()                                                                                          

In [36]: w2 = WCS() # "improved"                                                                             

In [37]: nd1 = NDData(np.arange(10), wcs=w1)                                                                 

In [38]: nd1.wcs = w2                                                                                        
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-38-0fc447b04304> in <module>
----> 1 nd1.wcs = w2

AttributeError: can't set attribute
```

Yes, I have found another workaround (besides the use of `nd1._wcs = w2`):

In [39]: nd2 = NDData(nd1, wcs=w2)                                                                           
INFO: overwriting NDData's current wcs with specified wcs. [astropy.nddata.nddata]

In [40]: nd2.wcs is w2                                                                                       
Out[40]: True
```

... but I cannot get rid of the unnecessary "info" message (at least not easily).

**Comment by pllim** [comment]
You can `log.setLevel("INFO")` again right after you assign that pesky WCS. 😉

**Comment by bsipocz** [comment]
@mwcraig - hasn't this been now addressed? Or I misunderstand the very end of the conversation at #9067 ?

**Comment by mwcraig** [comment]
Closed by #9067
